src/chart/legend.js

```diff
@@ -0,0 +1,91 @@
+(function(d3, fc) {
+    'use strict';
+
+    fc.chart.legend = function() {
+        var tableDecorate = fc.util.fn.noop,
+            rowDecorate = fc.util.fn.noop;
+
+        var items = [
+                ['datum', function(d) { return d.datum; }]
+            ];
+
+        var tableDataJoin = fc.util.dataJoin()
+            .selector('table.legend')
+            .element('table')
+            .attrs({'class': 'legend'});
+
+        var rowDataJoin = fc.util.dataJoin()
+            .selector('tr.row')
+            .element('tr')
+            .attrs({'class': 'row'});
+
+        var legend = function(selection) {
+            selection.each(function(data) {
+                var container = d3.select(this);
+
+                var legendData = items.map(function(item, i) {
+                    return [
+                        {
+                            value: d3.functor(item[0]),
+                            index: i,
```

Comment:
I think because of the data-join changes you'll be ok to just use the d3 supplied index?